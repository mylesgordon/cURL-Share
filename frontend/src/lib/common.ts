export function splitAndTrim(input: string): Array<string> {
		return input.split(',').map((item) => item.trim());
	}