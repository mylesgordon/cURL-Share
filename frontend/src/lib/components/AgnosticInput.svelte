<script lang="ts">
	export let id: string;
	export let label: string;
	export let value: string;
</script>

<div class="flex flex-col flex-grow overflow-hidden">
	<label for={id}>{label}</label>
	<input {id} class="input" bind:value {...$$props} />
</div>

<style>
	/* There is an issue with undefined bindings with agnostic ui's svelte implementation.
	This is just a direct copy of the input, input-rounded and label css styles */
	.input {
		color: var(--agnostic-font-color, var(--agnostic-dark));
		font-family: var(--agnostic-font-family-body);
		font-weight: var(--agnostic-font-weight, 300);
		font-size: var(--agnostic-font-size, 1rem);
		line-height: var(--agnostic-line-height, var(--fluid-20, 1.25rem));
		width: 100%;
		max-width: 100%;
		border-style: solid;
		border-width: var(--agnostic-input-border-size, 1px);
		border-color: var(--agnostic-input-border-color, var(--agnostic-gray-light));
		padding-block-start: var(--agnostic-input-vertical-pad, 0.5rem);
		padding-block-end: var(--agnostic-input-vertical-pad, 0.5rem);
		padding-inline-start: var(--agnostic-input-side-padding, 0.75rem);
		padding-inline-end: var(--agnostic-input-side-padding, 0.75rem);
		transition-property: box-shadow;
		transition-duration: var(--agnostic-input-timing, var(--agnostic-timing-medium));
		display: inline-block;
		margin-block-start: 0;
		margin-inline-start: 0;
		margin-inline-end: 0;
		/* margin-block-end: var(--agnostic-input-label-pad, 0.375rem); */
		vertical-align: initial;
		border-radius: var(--agnostic-radius, 0.25rem);
	}
</style>
