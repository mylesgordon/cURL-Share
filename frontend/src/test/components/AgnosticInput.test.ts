import '@testing-library/jest-dom/extend-expect';
import { get, writable } from 'svelte/store';
import { render, screen } from '@testing-library/svelte';
import { testMatchingSnapshot } from '../common';
import AgnosticInput from '$lib/components/AgnosticInput.svelte';
import html from 'svelte-htm';
import userEvent from '@testing-library/user-event';
import type { ComponentProps } from 'svelte';

describe('AgnosticInput component', () => {
	const props: ComponentProps<AgnosticInput> = {
		id: 'unit-test',
		label: 'Unit Test',
		value: 'Input'
	};
	testMatchingSnapshot(AgnosticInput, props);

	it('updates state correctly', async () => {
		const value = writable('Input');
		render(html`<${AgnosticInput} id="unit-test" label="Unit Test" bind:value=${value} />`);

		const input = await screen.findByLabelText('Unit Test');
		await userEvent.type(input, ' - now updated!');

		expect(get(value)).toBe('Input - now updated!');
	});
});
