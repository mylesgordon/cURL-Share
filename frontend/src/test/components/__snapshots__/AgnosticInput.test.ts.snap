// Vitest Snapshot v1

exports[`AgnosticInput component > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-col flex-grow overflow-hidden s-Et3LeF16QMsX"
    >
      <label
        class="s-Et3LeF16QMsX"
        for="unit-test"
      >
        Unit Test
      </label>
       
      <input
        class="input s-Et3LeF16QMsX"
        id="unit-test"
        label="Unit Test"
      />
    </div>
    <!--&lt;AgnosticInput&gt;-->
  </div>
</body>
`;
